.github/workflows/test-and-build.yml

```diff
@@ -0,0 +1,61 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Test and build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: 
+      - main
+
+jobs:
+  test-typecheck-lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js 16
+      uses: actions/setup-node@v2
+      with:
+        node-version: '16'
+        cache: 'npm'
+    - name: Restore packages
+      run: npm ci
+    - name: Run unit tests
+      run: npm test
+    - name: Run typecheck
+      run: npm run typecheck
+    - name: Run lint
+      run: npm run lint
+  
+  build-docker-image:
+    runs-on: ubuntu-latest
+    needs: [test-typecheck-lint]
+    # if: ${{ github.ref == 'refs/heads/main' }}
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js 16
+      uses: actions/setup-node@v2
+      with:
+        node-version: '16'
+        cache: 'npm'
+    - name: Restore packages
+      run: npm ci
+    - name: Build production assets
+      run: npm run build
+    - name: Build Docker image
+      run: docker build -t image .
+    - name: Log into registry
+      run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+    - name: Push image
+      run: |
+        TAG=1.0.${{ github.run_number }}  
```

Comment:
Nem lenne szerencsésebb, ha minden módosításkor csinálnánk itt githubon release-eket és az alapján lenne buildelve és tagelve az image? Semverrel kompatibilis lenne és valamennyire kikényszerítené a changelog vezetését is, ami esetleg a backend-frontend kooperációt is segítheti.

Reply:
Szerintem felesleges overhead kulon releaseket csinalni. Arra gondoltam, hogy a main branchbol menjen ki egy docker image, amit majd manual gombnyomasra kiteszunk staging kornyezetbe, es ha ott abban a kornyezetben a Selenium tesztek jok, akkor mehet elesbe. A backenddel meg tudunk koordinalni, hogy kompatibilisek maradjunk.

A semverezes egy npm vagy pip package-nel hasznos, de itt mar vegtermeket gyartunk, amit masok nem fognak felhasznalni, mint shared libraryt.

Ha esetleg a jovoben ez a pipeline nem felelne meg, akkor tudunk meg valtoztatni rajta.